### PR TITLE
docs: add an example on how to use custom parser with flat config

### PR DESCRIPTION
**Pull request by Lehoczky**
Related to: https://github.com/vuejs/eslint-plugin-vue/issues/2448

📚 Description

This PR adds an example on how to use this plugin with TypeScript, which is a common use case.

📜 Notes

The getting started guide is getting a bit hard to follow with a lot of warnings, duplicated text for each configuration style. It might worth extracting the `.eslintrc` format to a different subpage like it is done in [the official eslint docs](https://eslint.org/docs/latest/use/configure/configuration-files).

### CODE REVIEW

File: docs/user-guide/index.md [hunk 1]
```diff
@@ -146,6 +146,48 @@ If you want to use custom parsers such as [@babel/eslint-parser](https://www.npm
   }
 ```
 
+Full example:
+
+::: code-group
+
+```json [Legacy Config]
```
Comment (ota-meshi) [inline docs/user-guide/index.md:153]: Thanks for the PR!
I think we'll avoid using the terms "Legacy Config" and "Flat Config" in our documentation.
Could you please change it to use `.eslintrc` and `eslint.config.js` instead?

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: docs/user-guide/index.md (hunk 1 above)
Comment (Lehoczky) [reply docs/user-guide/index.md]: Sure thing! I saw the same naming convention elsewhere in the docs, I changed those too^^

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!
